src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/network-request-success.mdx

```diff
@@ -0,0 +1,1438 @@
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$url`
+      </td>
+
+            <td>
+        `string`
+      </td>
+
+      <td>
+        Required. The URL of the request.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$httpMethod`
+      </td>
+
+     <td>
+        `string`
+      </td>
+      <td>
+        Required. The HTTP method used, such as GET or POST.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$statusCode`
+      </td>
+
+      <td>
+        `int`
+      </td>
+      <td>
+        Required. The statusCode of the HTTP response, such as 200 for **OK**.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$startTime`
+      </td>
+
+      <td>
+        `int`
+      </td>
+
+      <td>
+        Required. The start time of the request in milliseconds since the epoch.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$endTime`
+      </td>
+
+      <td>
+        `int`
+      </td>
+      <td>
+        Required. The end time of the request in milliseconds since the epoch.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$bytesSent`
+      </td>
+
+      <td>
+        `int`
+      </td>
+      <td>
+        Required. The number of bytes sent in the request.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$bytesReceived`
+      </td>
+
+      <td>
+        `int`
+      </td>
+      <td>
+        Required. The number of bytes received in the response.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$responseBody`
+      </td>
+
+      <td>
+        `string`
+      </td>
+      <td>
+        Optional. The response body of the HTTP response. The response body will be truncated and included in an HTTP Error metric if the HTTP transaction is an error.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Examples [#examples]
+
+Here's an example of tracking an HTTP transaction:
+
+### Java [#java]
+
+```java
+public class CustomHttpMetricsLogger implements Interceptor {
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        Request request = chain.request();
+        //collect request start time
+        long t1 = System.nanoTime();
+        //get the size of the request body
+        long requestSize = null == request.body() ? 0 : request.body().contentLength();
+        //proceed with the request
+        Response response = chain.proceed(request);
+        //capture the time when response returns
+        long t2 = System.nanoTime();
+        long responseSize = null == response.body() ? 0 : response.body().contentLength();
+        //tell New Relic to notice this request
+        NewRelic.noticeHttpTransaction(request.urlString(), request.method(), response.code(), t1, t2, requestSize, responseSize);
+        //return response for processing
+        return response;
+    }
+}
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
```suggestion
```kotlin

class CustomInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val t1 = System.nanoTime()
        val aRequest: Request = chain.request()
        val requestSize: Long = if (null == aRequest.body) 0 else aRequest.body!!.contentLength()
        val aResponse = chain.proceed(aRequest)
        val t2 = System.nanoTime()
        val responseSize: Long = if (null == aResponse.body) 0 else aResponse.body!!.contentLength()
        NewRelic.noticeHttpTransaction(aRequest.url.toString(), aRequest.method, aResponse.code, t1, t2, requestSize, responseSize)
        return aResponse
    }
}

```
```